src/test/java/org/apache/commons/compress/harmony/pack200/Compress628Test.java

```diff
@@ -29,8 +29,8 @@ class Compress628Test {
     void test() throws Exception {
         final CPUTF8 name = new CPUTF8("");
         final CPUTF8 layout = new CPUTF8("Re\\T");
-        assertDoesNotThrow(() -> new NewAttributeBands(1, null, null,
-                new AttributeDefinitionBands.AttributeDefinition(35, AttributeDefinitionBands.CONTEXT_CLASS, name, layout)));
+        assertThrows(Pack200Exception.class, () -> new NewAttributeBands(1, null, null, new AttributeDefinitionBands.AttributeDefinition(35,
+                AttributeDefinitionBands.CONTEXT_CLASS, name, layout)));
```

Comment:
[COMPRESS-628](https://issues.apache.org/jira/browse/COMPRESS-628) was also about an OOM, so the change in behavior should not be relevant.

The native C implementation should also fail on the first unrecognized letter (`e`).